lib/prepare-driver.js

```diff
@@ -16,53 +16,67 @@ module.exports = function (app){
     var adapterModuleName = adapterDef.moduleName;
     var pathToAdapterDependency = adapterDef.modulePath;
     var adapterConfig = adapterDef.config;
+    var adapterClientName = adapterConfig.adapterClientName || 'redis';
+    var adapterClientConfig = adapterConfig.adapterClientConfig || {};
 
     // For redis:
     // ============================================================
     if (adapterModuleName === 'socket.io-redis') {
 
       // Create raw redis clients if necessary
-      var rawClientsNecessary = adapterConfig.pass || adapterConfig.db || adapterDef.adminBus;
+      var rawClientsNecessary = adapterConfig.pass || adapterConfig.db || adapterDef.adminBus || adapterConfig.adapterClientConfig;
       if (!rawClientsNecessary) {
         return cb();
       }
 
-      // Borrow access to the `redis` module from socket.io-redis
+      // Borrow access to the adapter redis client module from socket.io-redis
       var redis;
       try {
-        redis = require(path.resolve(pathToAdapterDependency, 'node_modules/redis'));
+        redis = require(adapterClientName);
       }
       catch (e) {
         try {
-          redis = require('redis');
+          redis = require(path.resolve(pathToAdapterDependency, 'node_modules/redis'));
         } catch (e2) {
-          return cb(e2);
+          try {
+            redis = require('redis');
+          } catch (e3) {
+            return cb(e3);
+          }
         }
       }
 
-      // Set up object that will be used for redis client options
-      var redisClientOpts = {};
```

Comment:
This will now be at the top (see other comments) -- and let's please preserve the code comment:

```js
 // Set up object that will be used for redis client options
```